diff --git a/src/bind.zsh b/src/bind.zsh
index 1dde137..4305a33 100644
--- a/src/bind.zsh
+++ b/src/bind.zsh
@@ -51,8 +51,16 @@ _zsh_autosuggest_bind_widget() {
 	# correctly. $WIDGET cannot be trusted because other plugins call
 	# zle without the `-w` flag (e.g. `zle self-insert` instead of
 	# `zle self-insert -w`).
+	# Preserve the ZLE_KILL | ZLE_YANK flags for builtin widgets for ZSH >= 5.2
 	eval "_zsh_autosuggest_bound_${bind_count}_${(q)widget}() {
 		_zsh_autosuggest_widget_$autosuggest_action $prefix$bind_count-${(q)widget} \$@
+		if [[ ! "${ZSH_VERSION}" < 5.2 ]]; then
+			case ${(q)widget} in
+				(${(j:|:)ZSH_AUTOSUGGEST_ZLE_KILL_WIDGETS}) zle -f 'kill';;
+				(${(j:|:)ZSH_AUTOSUGGEST_ZLE_YANK_WIDGETS}) zle -f 'yank';;
+				(${(j:|:)ZSH_AUTOSUGGEST_ZLE_YANKBEFORE_WIDGETS}) zle -f 'yankbefore';;
+			esac
+		fi
 	}"
 
 	# Create the bound widget
diff --git a/src/config.zsh b/src/config.zsh
index 5a0ebd8..6b45647 100644
--- a/src/config.zsh
+++ b/src/config.zsh
@@ -82,12 +82,41 @@ typeset -g ZSH_AUTOSUGGEST_ORIGINAL_WIDGET_PREFIX=autosuggest-orig-
 		run-help
 		set-local-history
 		which-command
-		yank
 		yank-pop
 		zle-\*
 	)
+	if [[ "${ZSH_VERSION}" < 5.2 ]]; then
+		ZSH_AUTOSUGGEST_IGNORE_WIDGETS+=(yank)
+	fi
 }
 
 # Pty name for capturing completions for completion suggestion strategy
 (( ! ${+ZSH_AUTOSUGGEST_COMPLETIONS_PTY_NAME} )) &&
 typeset -g ZSH_AUTOSUGGEST_COMPLETIONS_PTY_NAME=zsh_autosuggest_completion_pty
+
+# Widgets that should preserve ZLE_KILL flag
+(( ! ${+ZSH_AUTOSUGGEST_ZLE_KILL_WIDGETS} )) && {
+	typeset -ga ZSH_AUTOSUGGEST_ZLE_KILL_WIDGETS
+	ZSH_AUTOSUGGEST_ZLE_KILL_WIDGETS=(
+		kill-\*
+		backward-kill-\*
+	)
+}
+
+# Widgets that should preserve ZLE_YANK flag
+(( ! ${+ZSH_AUTOSUGGEST_ZLE_YANK_WIDGETS} )) && {
+	typeset -ga ZSH_AUTOSUGGEST_ZLE_YANK_WIDGETS
+	ZSH_AUTOSUGGEST_ZLE_YANK_WIDGETS=(
+		bracketed-paste
+		vi-put-after
+		yank
+	)
+}
+
+# Widgets that should preserve ZLE_YANKBEFORE flag
+(( ! ${+ZSH_AUTOSUGGEST_ZLE_YANKBEFORE_WIDGETS} )) && {
+	typeset -ga ZSH_AUTOSUGGEST_ZLE_YANKBEFORE_WIDGETS
+	ZSH_AUTOSUGGEST_ZLE_YANKBEFORE_WIDGETS=(
+		vi-put-before
+	)
+}
diff --git a/zsh-autosuggestions.zsh b/zsh-autosuggestions.zsh
index b19cac7..00fcc86 100644
--- a/zsh-autosuggestions.zsh
+++ b/zsh-autosuggestions.zsh
@@ -108,16 +108,45 @@ typeset -g ZSH_AUTOSUGGEST_ORIGINAL_WIDGET_PREFIX=autosuggest-orig-
 		run-help
 		set-local-history
 		which-command
-		yank
 		yank-pop
 		zle-\*
 	)
+	if [[ "${ZSH_VERSION}" < 5.2 ]]; then
+		ZSH_AUTOSUGGEST_IGNORE_WIDGETS+=(yank)
+	fi
 }
 
 # Pty name for capturing completions for completion suggestion strategy
 (( ! ${+ZSH_AUTOSUGGEST_COMPLETIONS_PTY_NAME} )) &&
 typeset -g ZSH_AUTOSUGGEST_COMPLETIONS_PTY_NAME=zsh_autosuggest_completion_pty
 
+# Widgets that should preserve ZLE_KILL flag
+(( ! ${+ZSH_AUTOSUGGEST_ZLE_KILL_WIDGETS} )) && {
+	typeset -ga ZSH_AUTOSUGGEST_ZLE_KILL_WIDGETS
+	ZSH_AUTOSUGGEST_ZLE_KILL_WIDGETS=(
+		kill-\*
+		backward-kill-\*
+	)
+}
+
+# Widgets that should preserve ZLE_YANK flag
+(( ! ${+ZSH_AUTOSUGGEST_ZLE_YANK_WIDGETS} )) && {
+	typeset -ga ZSH_AUTOSUGGEST_ZLE_YANK_WIDGETS
+	ZSH_AUTOSUGGEST_ZLE_YANK_WIDGETS=(
+		bracketed-paste
+		vi-put-after
+		yank
+	)
+}
+
+# Widgets that should preserve ZLE_YANKBEFORE flag
+(( ! ${+ZSH_AUTOSUGGEST_ZLE_YANKBEFORE_WIDGETS} )) && {
+	typeset -ga ZSH_AUTOSUGGEST_ZLE_YANKBEFORE_WIDGETS
+	ZSH_AUTOSUGGEST_ZLE_YANKBEFORE_WIDGETS=(
+		vi-put-before
+	)
+}
+
 #--------------------------------------------------------------------#
 # Utility Functions                                                  #
 #--------------------------------------------------------------------#
@@ -181,8 +210,16 @@ _zsh_autosuggest_bind_widget() {
 	# correctly. $WIDGET cannot be trusted because other plugins call
 	# zle without the `-w` flag (e.g. `zle self-insert` instead of
 	# `zle self-insert -w`).
+	# Preserve the ZLE_KILL | ZLE_YANK flags for builtin widgets for ZSH >= 5.2
 	eval "_zsh_autosuggest_bound_${bind_count}_${(q)widget}() {
 		_zsh_autosuggest_widget_$autosuggest_action $prefix$bind_count-${(q)widget} \$@
+		if [[ ! "${ZSH_VERSION}" < 5.2 ]]; then
+			case ${(q)widget} in
+				(${(j:|:)ZSH_AUTOSUGGEST_ZLE_KILL_WIDGETS}) zle -f 'kill';;
+				(${(j:|:)ZSH_AUTOSUGGEST_ZLE_YANK_WIDGETS}) zle -f 'yank';;
+				(${(j:|:)ZSH_AUTOSUGGEST_ZLE_YANKBEFORE_WIDGETS}) zle -f 'yankbefore';;
+			esac
+		fi
 	}"
 
 	# Create the bound widget
